.github/workflows/build.yml

```diff
@@ -0,0 +1,65 @@
+name: "Linting and Testing"
+
+on:
+  push:
+    branches: [master,github-actions-test]
+    paths-ignore:
+    - 'README.md'
+    - '.deepsource.toml'
+    - '.gitignore'
+
+  pull_request:
+    branches: [master]
+    paths-ignore:
+    - 'README.md'
+    - '.deepsource.toml'
+    - '.gitignore'
+
+
+jobs:
+  build:
+
+    runs-on: ${{matrix.os}}
+    strategy:
+      matrix:
+        os: [ubuntu-18.04]
+        python-version: [3.7,3.8,3.9]
+        test-folder : ['tests.tests_iotools']
+      fail-fast: false
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Build using Python ${{matrix.python-version}}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{matrix.python-version}}
+
+      - name: cache conda
+        uses: actions/cache@v1
+        with:
+          path: $CONDA
+          key: ${{ runner.os }}-conda-${{ hashFiles('iospi.yml') }}
+          restore-keys: |
+            ${{ runner.os }}-conda-
+      - name: install dependencies
+        run:  |
+          # $CONDA is an environment variable pointing to the root of the miniconda directory
+          $CONDA/bin/conda env update --file iospi.yml --name base
```

Comment:
ioSPI is sometimes called iospi and someting ioSPI: do we want to uniformize this? it could lead to weird (human or computer) errors...

Reply:
hmmm right this could be annoying going forward... PEP8 naming conventions require naming modules with all-lowercases (https://www.python.org/dev/peps/pep-0008/#package-and-module-names). I think it is a good idea, we will need to rename all the repos like that: ioSPI ==> iospi, reduceSPI ==> reducespi, etc. 
I am going to make sure all the other comments are adressed before I try to rename the repo.

Reply:
I think this could be addressed in another PR. I created an issue for now: https://github.com/compSPI/ioSPI/issues/5